src/DeviceInput/InputDevices/nativeDeviceInputSystem.ts

```diff
@@ -1,34 +1,23 @@
+import { INative } from "../../Engines/Native/nativeInterfaces";
 import { IUIEvent } from "../../Events/deviceInputEvents";
 import { Nullable } from "../../types";
 import { DeviceEventFactory } from "../Helpers/eventFactory";
 import { DeviceType, NativePointerInput, PointerInput } from "./deviceEnums";
 import { IDeviceInputSystem, INativeInput } from "./inputInterfaces";
 
+declare const _native: INative;
+
 /** @hidden */
 export class NativeDeviceInputSystem implements IDeviceInputSystem {
-    public onDeviceConnected = (deviceType: DeviceType, deviceSlot: number) => { };
-    public onDeviceDisconnected = (deviceType: DeviceType, deviceSlot: number) => { };
-    public onInputChanged = (deviceType: DeviceType, deviceSlot: number, eventData: IUIEvent) => { };
-
     private readonly _nativeInput: INativeInput;
 
-    public constructor(nativeInput?: INativeInput) {
-        this._nativeInput = nativeInput || this._createDummyNativeInput();
-
-        this._nativeInput.onDeviceConnected = (deviceType, deviceSlot) => {
-            this.onDeviceConnected(deviceType, deviceSlot);
-        };
-
-        this._nativeInput.onDeviceDisconnected = (deviceType, deviceSlot) => {
-            this.onDeviceDisconnected(deviceType, deviceSlot);
-        };
-
-        this._nativeInput.onInputChanged = (deviceType, deviceSlot, inputIndex, currentState) => {
+    public constructor(onDeviceConnected: (deviceType: DeviceType, deviceSlot: number) => void, onDeviceDisconnected: (deviceType: DeviceType, deviceSlot: number) => void, onInputChanged: (deviceType: DeviceType, deviceSlot: number, eventData: IUIEvent) => void) {
+        this._nativeInput = (_native.DeviceInputSystem) ? new _native.DeviceInputSystem(onDeviceConnected, onDeviceDisconnected, (deviceType, deviceSlot, inputIndex, currentState) => {
             const idx = (inputIndex === NativePointerInput.Horizontal || inputIndex === NativePointerInput.Vertical || inputIndex === NativePointerInput.DeltaHorizontal || inputIndex === NativePointerInput.DeltaVertical) ? PointerInput.Move : inputIndex;
             const evt = DeviceEventFactory.CreateDeviceEvent(deviceType, deviceSlot, idx, currentState, this);
 
-            this.onInputChanged(deviceType, deviceSlot, evt);
-        };
+            onInputChanged(deviceType, deviceSlot, evt);
+        }) : this._createDummyNativeInput();

```

Comment:
Missed this earlier. `_createDummyNativeInput` should be updated to not include the removed events.

Reply:
Removed from function and removed INativeInput